components/02-molecules/modal/modal.js

```diff
@@ -0,0 +1,150 @@
+    const focusableElements =
+      'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])';
+
+    // Check if an element is completely empty.
+    function isEmpty(node) {
+      return node.innerHTML.trim() === '';
+    }
+
+    // Toggle aria-[anything] attribute from true / false.
+    function toggleAria(el, aria) {
+      let x = el.getAttribute(`aria-${aria}`);
+      if (x === 'true') {
+        x = 'false';
+      } else {
+        x = 'true';
+      }
+      el.setAttribute(`aria-${aria}`, x);
+    }
+
+    // Traps keyboard focus when modal is open for ADA compliance
+    function trapKeyboard(modal) {
+      const firstFocusableElement =
+        modal.querySelectorAll(focusableElements)[0];
+      const focusableContent = modal.querySelectorAll(focusableElements);
+      const lastFocusableElement =
+        focusableContent[focusableContent.length - 1];
+
+      modal.addEventListener('keydown', (e) => {
+        const isTabPressed = e.key === 'Tab' || e.keyCode === 9;
+
+        if (!isTabPressed) {
+          return;
+        }
+
+        if (e.shiftKey) {
+          if (document.activeElement === firstFocusableElement) {
+            lastFocusableElement.focus();
+            e.preventDefault();
+          }
+        } else if (document.activeElement === lastFocusableElement) {
+          firstFocusableElement.focus();
+          e.preventDefault();
+        }
+      });
+
+      firstFocusableElement.focus();
+    }
+
+    // Toggle tabindex focus attribute.
+    function toggletabIndex(el) {
+      let x = el.getAttribute('tabindex');
+      if (x === '-1') {
+        x = '1';
+      } else {
+        x = '-1';
+      }
+      el.setAttribute('tabindex', x);
+    }
+
+    // Toggle Modal
+    function toggleModal(modal) {
+      const overlay = modal.querySelector('.modal__overlay');
+      const modalModalPane = modal.querySelector('.modal__pane');
+      if (overlay) {
+        overlay.classList.toggle('modal__overlay--active');
+      }
+      modal.classList.toggle('modal--active');
+      activeModal = context.querySelector('.modal--active');
+      toggleAria(modalModalPane, 'hidden');
+      toggleAria(modal, 'expanded');
+      trapKeyboard(modalModalPane);
+      toggletabIndex(modalModalPane);
+    }
+
+    // AJAX Request data from url + CSS selector.
+    function ajaxRequest(request, url, selector) {
+      let response = '';
+      request.onreadystatechange = function processResponse() {
+        if (request.status === 200) {
+          const resp = request.responseText;
+          const parser = new DOMParser();
+          const htmlDoc = parser.parseFromString(resp, 'text/html');
+          const remoteselector = htmlDoc.querySelectorAll(selector);
+
+          response = remoteselector[0].innerHTML;
+        } else {
+          response = `<p>Could not process request. Error code: ${request.status}</p>`;
+        }
+      };
+      request.open('GET', url, false);
```

Comment:
Why are we including ajax functionality for the modal? This seems very strange to me. The modern standard is `fetch`, some people still prefer `axios`, and even fetching data from a modal seems like an edge case.

We should remove this. Kudos for trying to cover all the bases, but let's let the small amount of people doing this decide how they want to implement fetching data.